src/migrator.coffee

```diff
@@ -33,369 +23,380 @@ isPositiveInteger = (value) ->
     return false
   return true
 
+validateOptions = (optionsToCheck) ->
+  # don't check the 'logging' value here, it will check itself elsewhere
+
+  if optionsToCheck["on-read-error"]? && ["ignore", "log", "exit"].indexOf(optionsToCheck["on-read-error"]) == -1
+    throw "invalid value for 'on-read-error': '#{optionsToCheck["on-read-error"]}'"
+
+  if optionsToCheck["command-buffering"]? && !isPositiveInteger(optionsToCheck["command-buffering"])
+    throw "positive integer required for 'command-buffering', got: '#{optionsToCheck["command-buffering"]}'"
+
+  if optionsToCheck["migration-at-once"] && optionsToCheck["autocommit"]
+    throw "'migration-at-once' and 'autocommit' may not be used together"
+
+  if optionsToCheck["one-transaction"] && optionsToCheck["autocommit"]
+    throw "'one-transaction' and 'autocommit' may not be used together"
+
+  if optionsToCheck.order? && ["basename", "path"].indexOf(optionsToCheck.order) == -1
+    throw "invalid value for 'order': '#{optionsToCheck.order}'"
+
+    
+defaults =
+  table: 'dbsync_migrations'
+  files: '*.sql'
+  logging: 'info'
+  order: 'path'
+  "on-read-error": 'exit'
+  "command-buffering": 4
+  encoding: 'utf8'
+
+
 module.exports = class Migrator
-  _$ = {}
+
+  # note: this is a static property of the class, not an instance property
+  @defaults: defaults
 
   constructor: (_options) ->
-    bindPrivateMethods(_$, this)
     @name = "Migrator"
-    _$.validateOptions(_options) # throws if there is a problem
-    _$.options = _.clone(_options)
-    _.defaults(_$.options, defaults)
-    _$.logger = loggerFactory.getLogger(_$.options.logging)
-    _$.logger.debug "STATUS: received migration _$.options: #{JSON.stringify(_$.options,null,2)}"
+    validateOptions(_options) # throws if there is a problem
+    options = _.clone(_options)
+    _.defaults(options, defaults)
+    logger = loggerFactory.getLogger(options.logging)
+    logger.debug "STATUS: received migration options: #{JSON.stringify(options,null,2)}"
```

Comment:
Just made private variables part of the constructor closure, and moved public methods into constructor as well so they get access to the private/closured variables.  It's a shame this was necessary, because it doesn't use `prototype` this way, and thus "inherited" public methods show up as "own" properties.  But, at least the code is simpler this way.